### PR TITLE
ALBS-1044: Add build flavor repos when scheduling test tasks

### PR DESCRIPTION
**Pull request by javihernandez**

### CODE REVIEW

**Review by soksanichenko — APPROVED** [review]

**Review by maccelf — APPROVED** [review]

File: alws/test_scheduler.py [hunk 1]
```diff
@@ -22,6 +22,49 @@ def __init__(self, term_event: threading.Event,
         self._graceful_event = graceful_event
         self._alts_client = AltsClient(settings.alts_host, settings.alts_token)
 
+
+    @staticmethod
+    def __get_repos_for_test_task(task: models.TestTask):
```
Comment (anfimovdm) [inline alws/test_scheduler.py:27]: please, clarify return type

File: alws/test_scheduler.py [hunk 2]
```diff
@@ -22,6 +22,49 @@ def __init__(self, term_event: threading.Event,
...
+                    '$releasever',
+                    platform.distr_version
+                )
+            }
+            for flavor in task.build_task.build.platform_flavors
+            for item in flavor.repos
+            if item.type == 'rpm'
+            and item.arch == task.env_arch
+        ]
+        repos.extend(flavor_repos)
```
Comment (anfimovdm) [inline alws/test_scheduler.py:63]: can it be better, if we will iterate by new arrays and add them in `repos`?
like this
```python3
build_repositories = [...]
linked_build_repos = [...]
flavor_repos = [...]
for repo_arr in (
    build_repositories,
    linked_build_repos,
    flavor_repos,
):
    repos.extend(repo_arr)
```
it's up to you

**Review by anfimovdm — CHANGES_REQUESTED** [review]

**Comment by anfimovdm** [comment]
also, could you please create unit test for this code?

**Comment by javihernandez** [comment]
> also, could you please create unit test for this code?

Added unit tests for the newly introduced function and fixed review comments, ready again for another round of review

**Review by anfimovdm — APPROVED** [review]

**Pull request merged**